src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -251,6 +251,19 @@ public static String formatDateTimeForInstructorHomePage(Date date) {
         return sdf.format(date);
     }
 
+    /**
+     * Converts {@code date} to UTC according to the {@code timeZone} offset and formats it according to ISO8601.
+     */
+    public static String convertAndFormatDateToIso8601Utc(Date date, double timeZone) {
+        if (date == null) {
+            return "";
+        }
+        // Passing negative timeZone to convert to UTC.
+        Calendar calInUtc = convertToUserTimeZone(dateToCalendar(date), -timeZone);
```

Comment:
I believe you can now make use of your newly created method `convertLocalDateToUtc` :)